adding owners to Team

bnm91 opened the pull request:
## Problem
Currently there's no way to get info on the "owner" (aka "manager") of a Team.  While the team's name is available, those are subject to change making tracking over longer periods difficult

## Solution
ESPN has a concept of Members on a League and Owners on a Team.  These are tied together by an Id and the Member entity has much more stable information about the actual humans behind the team.
Storing this Member info with the Team will make that info more accessible

## Code Changes
* added members property to the League
* modified BaseLeague fetch_teams to create and an array of members associated with the owners of each Team
* added owners property and constructor parameter to Teams

bnm91 commented:
I’ll take a look at these failures soon

cwendt94 commented:
Don't worry about the hockey team names one those just popped up, but it does look like two new ones are failing for `owner` logic.

cwendt94 commented:
FYI I did have owners but them removed because they no longer had the actual name, but I can see why we would want at least the unique id. Here was the [PR](https://github.com/cwendt94/espn-api/commit/cc82818bada7bda0e12fb10688943fbb46103bd2#diff-7ca2b36d4c2c88bf07608e13b4d17ec8a4b6396ea413738563a137e731f0f6f2) removing it. There is a chance a team might not have a owner see this [comment](https://github.com/cwendt94/espn-api/commit/cc82818bada7bda0e12fb10688943fbb46103bd2#diff-7ca2b36d4c2c88bf07608e13b4d17ec8a4b6396ea413738563a137e731f0f6f2)

cwendt94 commented on espn_api/base_league.py:
```diff
@@ -92,6 +97,17 @@ def _get_all_pro_schedule(self):
+        for team in data['teams']:
+            team_owners[team['id']] = []
+            for member in self.members:
+                for owner in team['owners']:
```
team might not have an owner, I would suggest using `.get` to get `owners` filed and falling back to empty array

samthom1 commented:
For what it's worth, there may have been further changes to the ESPN API this morning. I was making my own attempt at fixing this issue when the API call started returning additional fields under data['members']. Restoring the members/owner references from before commit cc82818 allows owner names to flow through again.

bnm91 commented:
@cwendt94 I rewrote this to only include the id as discussed.  I also fixed a couple tests broken by my change.
I still think it would be nice to have the "member" (and therefore the user's actual name) associated with the Team but currently this leaves that exercise to the user.  Perhaps I'll open a separate discussion for that improvement


I'm struggling to run the entire test suite in my environment but addressed a few test issues I was aware of.

The following tests will probably still fail but due to separate issue:
test_league_draft (basketball.integration.test_league.LeagueTest)
	separate issue -- test uses 2019 data which requires 'location' but none exists
	
test_box_scores (football.integration.test_league.LeagueTest)
	separate issue -- test uses 2019 data which requires 'location' but none exists
	
test_league_int (hockey.integration.test_league.LeagueTest)
	separate issue -- test uses 2021 data which requires 'location' but none exists

cwendt94 commented on espn_api/base_league.py:
```diff
@@ -48,12 +50,15 @@ def _fetch_teams(self, data, TeamClass, pro_schedule = None):
         for team in teams:
             roster = team_roster[team['id']]
-            self.teams.append(TeamClass(team, roster=roster, schedule=schedule, year=seasonId, pro_schedule=pro_schedule))
+            owners = team_owners[team['id']]
```
You know what I just realized is the `team` data passing into the team class already has the owners data so we shouldn't need any of this logic in base_league. See comment in team class below

cwendt94 replied:
Im not sure why I added roster like this either when its also in the `team` data object. I will clean up that logic later

cwendt94 commented on espn_api/baseball/team.py:
```diff
@@ -5,7 +5,7 @@
     '''Teams are part of the league'''
-    def __init__(self, data, roster, schedule, year, **kwargs):
+    def __init__(self, data, roster, schedule, year, owners, **kwargs):
         self.team_id = data['id']
```
I think for each team class all you will need to add is `self.owners = data.get('owners', [])`

DesiPilla commented:
@bnm91 Is there an endpoint / header we can use to get the actual users' names?

DesiPilla commented:
I'm using the following workaround for now:
```python
def set_league_endpoint(league: League) -> None:
    """Set the league's endpoint."""

    # Current season
    if league.year >= (datetime.datetime.today() - datetime.timedelta(weeks=12)).year:
        league.endpoint = (
            "https://fantasy.espn.com/apis/v3/games/ffl/seasons/"
            + str(league.year)
            + "/segments/0/leagues/"
            + str(league.league_id)
            + "?"
        )

    # Old season
    else:
        league.endpoint = (
            "https://fantasy.espn.com/apis/v3/games/ffl/leagueHistory/"
            + str(league.league_id)
            + "?seasonId="
            + str(league.year)
            + "&"
        )

def set_owner_names(league: League):
    """This function sets the owner names for each team in the league.
    The team.owners attribute only contains the SWIDs of each owner, not their real name.

    Args:
        league (League): ESPN League object
    """
    endpoint = "{}view=mTeam".format(league.endpoint)
    r = requests.get(endpoint, cookies=league.cookies).json()
    if type(r) == list:
        r = r[0]

    # For each member in the data, create a map from SWID to their full name
    swid_to_name = {}
    for member in r["members"]:
        swid_to_name[member["id"]] = re.sub(
            " +", " ", member["firstName"] + " " + member["lastName"]
        ).title()

    # Set the owner name for each team
    for team in league.teams:
        team.owner = swid_to_name[team.owners[0]]

set_league_endpoint(league)
set_owner_names(league)
```
